feat: add build loading awareness in the terminal(#1117)

actopas opened the pull request:
## Details

This PR enhances the build process feedback by implementing a loading animation in the terminal. It addresses the issue of unclear build status during development, especially for longer builds or multiple hot reloads.

Specifically, this PR:
- Adds a loading animation at the start of each build/hot reload process
- Removes the animation when the build completes (success or failure)
- Improves visibility of the current build status

This change aims to solve the problem described in issue #1117, where it was difficult to determine if a build was in progress or completed, especially after multiple hot reloads.

I'm open to any feedback or suggestions for improvement. If you have any questions or if there's anything you'd like me to modify, please leave a comment.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/PlasmoHQ/plasmo/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I agree to license this contribution under the MIT LICENSE
- [x] I checked the [current PR](https://github.com/PlasmoHQ/plasmo/pulls) for duplication.

## Contacts

- (OPTIONAL) Discord ID: actopas

louisgv commented on cli/plasmo/src/features/helpers/loading-animation.ts:
This is pretty cool!!

louisgv replied:
The current convention is a bit extreme on not using "let" :-? 

Can we make a const state like this instead:

```ts
const state = {
  loadingInterval: null as NodeJS.Timeout | null,
  isLoading: false,
  dotCount: 0
}
```

louisgv commented on cli/plasmo/src/features/helpers/loading-animation.ts:
```diff
@@ -0,0 +1,26 @@
+  if (isLoading) return
+  isLoading = true
+  let dots = 0
+  const baseString = "🔄 Building"
```
let's pull this outside, since it's a literal string

louisgv commented on cli/plasmo/src/commands/dev.ts:
```diff
@@ -62,20 +63,29 @@ async function dev() {
+      return
+    }
+
+    if (event.type === "buildFailure") {
+      stopLoading()
```
nice!

louisgv commented on cli/plasmo/src/features/helpers/loading-animation.ts:
```diff
@@ -0,0 +1,26 @@
+let isLoading = false
+
+export const startLoading = () => {
+  if (isLoading) return
```
nit: let's use bracket

louisgv commented on cli/plasmo/src/features/helpers/loading-animation.ts:
```diff
@@ -0,0 +1,26 @@
+}
+
+export const stopLoading = () => {
+  if (!isLoading) return
```
nit: brakcet

louisgv reviewed (commented):
Looks like a pretty cool feature!

actopas commented:
@louisgv  Thanks for the review! I've addressed all the suggestions:

- ✅ Moved state management to a single `state` object
- ✅ Added consistent bracket style for all conditions
- ✅ Extracted `LOADING_TEXT` as a constant

Please let me know if there's anything else that needs to be adjusted.

louisgv reviewed (approved):
LGTM 👍
